Changed parsing of ESUTIL version to account for release candidates

jcbird opened the pull request:
Hi Jo,

This is super small. For another reason, I needed to compile the master branch of esutil. It's version is '0.6.2rc2' which gave an exception when trying to int('2rc2'). I looked at the code and realized that the second number (here, '6') was the one you were interested in, so I used the built-in translate to remove  'rc' from the string. Now, 
_ESUTIL_VERSION = [0, 6, 22]  for my esutil version. The code runs as it should.

Cheers,
Jon

jobovy commented:
Hi Jon,

Thanks for noticing and the fix. However, I don't think this is quite the correct parsing, because the 3rd version number in this case is ``2`` and not ``22`` (because I think this is the 2nd release candidate?). In that case, I think you want
```
_ESUTIL_VERSION= [int(v.split('rc')[0]) for v in esutil.__version__.split('.')]
```
?

jcbird commented:
Yep, sounds better, especially if esutil doesn't have incremental releases
above 10.   0.6.11 would be no good.

On Fri, Dec 16, 2016, 11:20 AM Jo Bovy <notifications@github.com> wrote:

> Hi Jon,
>
> Thanks for noticing and the fix. However, I don't think this is quite the
> correct parsing, because the 3rd version number in this case is 2 and not
> 22 (because I think this is the 2nd release candidate?). In that case, I
> think you want
>
> _ESUTIL_VERSION= [int(v.split('rc')[0]) for v in esutil.__version__.split('.')]
>
> ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jobovy/apogee/pull/41#issuecomment-267646204>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGIBk4P1JrEVaW4Rcft3wjNQRFSdbE8dks5rIshCgaJpZM4LOs09>
> .
>

jobovy commented:
Great, do you want to push that change to this branch and I'll merge it?

jcbird commented:
Sure, give me an hour or two. Running around at the moment. Thanks!

On Fri, Dec 16, 2016 at 11:35 AM Jo Bovy <notifications@github.com> wrote:

> Great, do you want to push that change to this branch and I'll merge it?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jobovy/apogee/pull/41#issuecomment-267650199>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGIBk4sP5Gb7lSaIuaORPCgtLp56_BPuks5rIsvFgaJpZM4LOs09>
> .
>
